build/assert-publish-aot-testApp-warning-count.ps1

```diff
@@ -0,0 +1,24 @@
+$rootDirectory = Split-Path $PSScriptRoot -Parent
+$publishOutput = dotnet publish $rootDirectory\test\OpenTelemetry.AotCompatibility.TestApp\OpenTelemetry.AotCompatibility.TestApp.csproj -nodeReuse:false /p:UseSharedCompilation=false
+
+$actualWarningCount = 0
+
+foreach ($line in $($publishOutput -split "`r`n"))
+{
+    if ($line -like "*analysis warning IL*") 
+    {
+        $actualWarningCount += 1
+    }
+}
+
+Write-Host "Actual warning count is:", $actualWarningCount
+$expectedWarningCount = 28
+
+$testPassed = 0
+if ($actualWarningCount -ne $expectedWarningCount)
+{
+    $testPassed = 1
+    Write-Host "Actual warning count:", actualWarningCount, "is not as expected. Expected warning count is:", $expectedWarningCount

```

Comment:
If the warning count doesn't match, do we need to reveal some details about what's the actual output or what's the mismatch?

Reply:
That's a good point. We should print out the warnings in any case.

Reply:
@Yun-Ting Consider doing this in a follow-up PR.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet/pull/4734